core/domain/html_validation_service.py

```diff
@@ -923,6 +923,87 @@ def check_for_math_component_in_html(html_string):
     return bool(math_tags)
 
 
+def extract_math_rich_text_related_information_from_html(html_string):
+    """Extract information from the math rich-text components from an HTML
+    string. This information will be useful for generating SVG images for these
+    math tags.
+
+    Args:
+        html_string: str. The HTML string.
+
+    Returns:
+        tuple(int, str, list). A 3-tuple whose elements are as follows:
+        - int. The approximate size of Math SVGs in bytes.
+        - str. The largest raw_latex value in the html string by length.
+        - list. All unique raw_latex values in the math tags.
+    """
+
+    soup = bs4.BeautifulSoup(
+        html_string.encode(encoding='utf-8'), 'html.parser')
+    size_in_bytes = 0
+    largest_math_expression = ''
+    # The approximate size for an SVG image for a Latex with one character
+    # is around 1000 Kb. But, when the number of characters increase the
+    # size of SVG per character reduces. For example: If the size of SVG
+    # for the character 'a' is 1000 bytes, the size of SVG for 'abc' will
+    # be lesser than 3000 bytes. So the below approximation to find the
+    # size will give us the maximum size.
+    latex_values = []
+    for math_tag in soup.findAll(name='oppia-noninteractive-math'):
+        math_content_dict = (
+            json.loads(unescape_html(
+                math_tag['math_content-with-value'])))
+        raw_latex = (
+            objects.UnicodeString.normalize(math_content_dict['raw_latex']))
+        latex_values.append(raw_latex)
+    unique_latex_values = list(set(latex_values))
+    for latex_value in unique_latex_values:
+        # The characters in special Latex keywords like 'frac' and 'sqrt' don't
+        # add up to the total size of SVG.
+        length_of_expression = (
+            len(
+                latex_value.replace('frac', '').replace('sqrt', '').replace(
+                    ' ', '')))
+        size_in_bytes = size_in_bytes + (length_of_expression * 1000)
+        largest_math_expression = (
```

Comment:
largest --> longest (given that you are comparing by length).

It might also be clearer to name this longest_raw_latex_string throughout. When reading it above I was wondering what format it was in, or whether it was a domain object.

Reply:
done